src/components/molecules/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,121 @@
+const MAX_HOUR = 24;
+const MIDDAY_HOUR = 12;
+
+export const Schedule: React.FC<ScheduleProps> = ({ scheduleDay }) => {
+  const getStartHour = useCallback(
+    (utcSeconds: number) => {
+      return utcSeconds >= scheduleDay.dayStartUtcSeconds
+        ? Number(format(fromUnixTime(utcSeconds), "H"))
+        : 0;
+    },
+    [scheduleDay]
+  );
+
+  const roomStartHour = useCallback(
+    (room: RoomWithEvents) =>
+      room.events?.reduce(
+        (acc, event) => Math.min(acc, getStartHour(event.start_utc_seconds)),
+        MAX_SCHEDULE_START_HOUR
+      ) ?? MAX_SCHEDULE_START_HOUR,
+    [getStartHour]
+  );
+
+  const scheduleStartHour = useMemo(
+    () =>
+      Math.min(
+        ...scheduleDay.rooms.map((room) => roomStartHour(room)),
+        MAX_SCHEDULE_START_HOUR
+      ),
+    [scheduleDay, roomStartHour]
+  );
+
+  const hours = useMemo(
+    () =>
+      range(scheduleStartHour, MAX_HOUR).map(
+        (hour: number) =>
+          `${hour % MIDDAY_HOUR || MIDDAY_HOUR} ${
+            hour >= MIDDAY_HOUR ? "PM" : "AM"
+          }`
+      ),
+    [scheduleStartHour]
+  );
+
+  const containerVars = useCss({
+    "--room-count": scheduleDay.rooms.length + 1,
+    "--current-time--position": calcStartPosition(
+      Math.floor(getUnixTime(Date.now())),
+      scheduleStartHour
+    ),
+    "--hours-count": hours.length,
+  });
+
+  const containerClasses = useMemo(
+    () => classNames("ScheduledEvents", containerVars),
+    [containerVars]
+  );
+
+  return (
+    <div className={containerClasses}>
+      <div className="ScheduledEvents__rooms">
+        <div className="ScheduledEvents__room">
+          <p className="ScheduledEvents__room-title">My Daily Schedule</p>
+          <span className="ScheduledEvents__events-count">
+            {scheduleDay.personalEvents.length} events
+          </span>
+        </div>
+        {scheduleDay.rooms.map((room) => (
+          <div key={room.title} className="ScheduledEvents__room">
+            <p className="ScheduledEvents__room-title">{room.title}</p>
+            <span className="ScheduledEvents__events-count">
+              {room.events?.length || 0} events
+            </span>
+          </div>
+        ))}
+      </div>
+
+      <div className="ScheduledEvents__schedule">
+        <div className="ScheduledEvents__timeline">
+          {hours.map((hour: string) => (
+            <span className="ScheduledEvents__hour" key={hour}>
+              {hour}
+            </span>
+          ))}
+        </div>
+
+        {scheduleDay.isToday && (
+          <div className="ScheduledEvents__current-time-line"></div>
+        )}
+
+        <div className="ScheduledEvents__user-schedule">
+          <ScheduleRoomEvents
+            isUsers
+            events={scheduleDay.personalEvents}
+            scheduleStartHour={scheduleStartHour}
+          />
+        </div>
+        {scheduleDay.rooms.map((room) => (
+          <ScheduleRoomEvents
+            key={room.title}
```

Comment:
the logic is updated. Now we have `venueId` and `roomTitle` as key: https://github.com/sparkletown/sparkle/commit/28b4cc25e369c09ac8ce1a6b2e3f4ee709068144

Reply:
See the comment I made somewhere else on a similar review thread about adding `index` + `@debt` comment like:

```
// @debt Ideally event.id would always be a unique identifier, but our types suggest it
//   can be undefined. Because we can't use index as a key by itself (as that is unstable
//   and causes rendering issues, we construct a key that, while not guaranteed to be unique,
//   is far less likely to clash
```

Reply:
see my answer there as well: https://github.com/sparkletown/sparkle/pull/1302#discussion_r633808911